src/llm/apis/openai_completions.cpp

```diff
@@ -128,6 +131,57 @@ ov::Tensor load_image_stbi(const std::string& imageBytes) {
         SharedImageAllocator{image, desiredChannels, y, x});
 }
 
+static size_t writeMemoryCallback(void* contents, size_t size, size_t nmemb,
+    void* userp) {
+    size_t realsize = size * nmemb;
+    auto& mem = *static_cast<std::string*>(userp);
+    mem.append(static_cast<char*>(contents), realsize);
+    return realsize;
+}
+
+static absl::Status downloadImage(const char* url, std::string& decoded, const double& sizeLimit) {
+    CURL* curl_handle = curl_easy_init();
+    auto status = curl_easy_setopt(curl_handle, CURLOPT_URL, url);
+    if (status == CURLE_OK) {
+        status = curl_easy_setopt(curl_handle, CURLOPT_WRITEFUNCTION, writeMemoryCallback);
+    }
+    if (status == CURLE_OK) {
+        status = curl_easy_setopt(curl_handle, CURLOPT_WRITEDATA, &decoded);
+    }
+    if (status == CURLE_OK) {
+        status = curl_easy_setopt(curl_handle, CURLOPT_SSL_OPTIONS, CURLSSLOPT_NATIVE_CA);
+    }
+    if (status == CURLE_OK) {
+        status = curl_easy_setopt(curl_handle, CURLOPT_TIMEOUT, 20L);
+    }
+    if (status == CURLE_OK) {
+        status = curl_easy_setopt(curl_handle, CURLOPT_FOLLOWLOCATION, 1L);
+    }
+    if (status == CURLE_OK) {
+        status = curl_easy_setopt(curl_handle, CURLOPT_HTTPPROXYTUNNEL, 1L);
+    }
+    if (status == CURLE_OK) {
+        status = curl_easy_setopt(curl_handle, CURLOPT_HTTPPROXYTUNNEL, 1L);
+    }
+    if (status == CURLE_OK) {
+        status = curl_easy_setopt(curl_handle, CURLOPT_MAXFILESIZE, sizeLimit);
+    }
+    if (status == CURLE_OK) {
+        status = curl_easy_perform(curl_handle);
+    }
+
+    if (status != CURLE_OK) {
+        std::stringstream ss;
+        ss << "Downloading image failed: " << curl_easy_strerror(status);
+        SPDLOG_LOGGER_ERROR(llm_calculator_logger, ss.str());
+        return absl::InvalidArgumentError(ss.str());
```

Comment:
You return the same message that we put in the logs. I don't know the contents of such errors, but wouldn't they reveal to much if we send them back to the client?